configure_reclient.py

```diff
@@ -0,0 +1,560 @@
+                                             Paths.build_dir),
+            'clang_include_dir': clang_include_dir,
+            'linux_clang_base_path': Paths.relpath(Paths.linux_clang_base_path,
+                                                   Paths.build_dir),
+            'linux_clang_include_dir': linux_clang_include_dir,
+        }
+
+        # Substitute variables into the template.
+        clang_remote_wrapper = ShellTemplate(
+            clang_remote_wrapper_template).substitute(template_vars)
+
+        # Write the clang remote wrapper.
+        FileUtils.write_text_file(
+            (f'{Paths.src_dir}/buildtools/reclient_cfgs/chromium-browser-clang/'
+             'clang_remote_wrapper'), clang_remote_wrapper)
+
+    def generate_reproxy_cfg(self):
+        # Load Chromium config template and remove everything starting with $
+        # symbol on each line.
+        reproxy_template_fname = 'reproxy_cfg_templates/reproxy.cfg.template'
+        reproxy_template_file = f'{Paths.reclient_cfgs_dir}/{reproxy_template_fname}'
+        if not os.path.isfile(reproxy_template_file):
+            reproxy_template_file = f'{Paths.script_dir}/{reproxy_template_fname}'
+        reproxy_cfg = ReclientCfg.parse_from_string(
+            re.sub(r'^([^$]+)\$.*$',
+                   r'\1',
+                   FileUtils.read_text_file(reproxy_template_file),
+                   flags=re.MULTILINE))
+
+        # Merge with our config.
+        source_cfg_paths = [
+            f'{Paths.script_dir}/reproxy.cfg',
+        ]
+        for source_cfg_path in source_cfg_paths:
+            reproxy_cfg = ReclientCfg.merge_cfg(reproxy_cfg, source_cfg_path)
+
+        # Use scandeps_server.
+        depsscanner_address = (f'exec://{Paths.src_dir}/'
+                               'buildtools/reclient/scandeps_server')
+        if sys.platform.startswith('win'):
+            depsscanner_address += '.exe'
+        reproxy_cfg['depsscanner_address'] = depsscanner_address
+
+        # Launch a custom merge step if it exists.
+        if self.custom_py and 'merge_reproxy_cfg' in self.custom_py:
+            reproxy_cfg = self.custom_py['merge_reproxy_cfg'](reproxy_cfg)
+            source_cfg_paths.append(Paths.custom_py)
+
+        # Write the final config to the expected location.
+        ReclientCfg.write_to_file(f'{Paths.reclient_cfgs_dir}/reproxy.cfg',
+                                  reproxy_cfg, source_cfg_paths)
+
+    def generate_rewrapper_cfgs(self):
+        for tool in ['chromium-browser-clang', 'python']:
+            for platform in ['linux', 'mac', 'windows']:
+                self.generate_rewrapper_cfg(tool, platform)
+
+    def generate_rewrapper_cfg(self, tool, host_os):
+        # Load Chromium config for linux remote.
+        rewrapper_cfg_fname = f'linux/{tool}/rewrapper_linux.cfg'
+        rewrapper_cfg_file = f'{Paths.reclient_cfgs_dir}/{rewrapper_cfg_fname}'
+        if not os.path.isfile(rewrapper_cfg_file):
```

Comment:
this may also fallback to old configs if Chromium decides to move things around in `main`. We need to be aware of this scenario, because things might continue working, but the resulting configs may become more and more outdated.

Reply:
Yep, agreed!